internal/components/multi_endpoint.go

```diff
@@ -72,6 +73,37 @@ func (m *MultiPortReceiver) ParserName() string {
 	return fmt.Sprintf("__%s", m.name)
 }
 
+func (m *MultiPortReceiver) GetDefaultConfig(logger logr.Logger, config interface{}) (interface{}, error) {
+	multiProtoEndpointCfg := &MultiProtocolEndpointConfig{}
+	if err := mapstructure.Decode(config, multiProtoEndpointCfg); err != nil {
+		return nil, err
+	}
+	tmp := make(map[string]*SingleEndpointConfig, len(multiProtoEndpointCfg.Protocols))
+	for protocol, ec := range multiProtoEndpointCfg.Protocols {
+		var port int32
+		if defaultSvc, ok := m.portMappings[protocol]; ok {
+			port = defaultSvc.Port
+			if ec != nil {
+				port = ec.GetPortNumOrDefault(logger, port)
+			}
+		}
+		var addr string
+		if defaultAddr, ok := m.addrMappings[protocol]; ok {
+			addr = defaultAddr
+		}
+		res, err := AddressDefaulter(logger, addr, port, ec)
+		if err != nil {
+			return nil, err
+		}
+		tmp[protocol] = res
+	}
+
+	for protocol, ec := range tmp {
+		multiProtoEndpointCfg.Protocols[protocol] = ec
+	}
+	return config, mapstructure.Decode(multiProtoEndpointCfg, &config)
```

Comment:
In case we need to marshal `listenaddress`, I think we should remove the entry here.